### PR TITLE
chore: remove ethers dependency from `tycho-storage`

### PR DESCRIPTION
**Pull request by zizou0x**
This PR aims to remove the `ethers` dependency from `tycho-storage` without changing its logic.

### CODE REVIEW

File: tycho-storage/src/postgres/mod.rs [hunk 1]
```diff
@@ -401,6 +402,19 @@ async fn maybe_lookup_version_ts(
...
+pub fn keccak256<T: AsRef<[u8]>>(bytes: T) -> [u8; 32] {
+    let mut output = [0u8; 32];
+
+    let mut hasher = Keccak::v256();
+    hasher.update(bytes.as_ref());
+    hasher.finalize(&mut output);
+
+    output
+}
+
```
Comment (zizou0x) [inline tycho-storage/src/postgres/mod.rs:417]: This is going to be used by `tycho-indexer` too. Should we introduce a `tycho-utils` crate for these shared utils functions or maybe directly in `tycho-core`?

Comment (zizou0x) [reply tycho-storage/src/postgres/mod.rs:417]: This was moved in `tycho-core` in my other PR, with an open question about `tycho-utils` still

Comment (louise-poole) [reply tycho-storage/src/postgres/mod.rs:417]: I think it's fine in tycho-core 👍

File: tycho-core/src/hex_bytes.rs [hunk 2]
```diff
@@ -226,6 +300,24 @@ macro_rules! impl_from_for_ethers_fixed_hash {
 
 impl_from_for_ethers_fixed_hash!(H160, H256);
 
+macro_rules! impl_from_uint_for_bytes {
```
Comment (louise-poole) [inline tycho-core/src/hex_bytes.rs:303]: Made your own macro - aren't you cool 🤓

File: tycho-storage/src/postgres/chain.rs [hunk 3]
```diff
@@ -582,16 +581,14 @@ mod test {
...
         let c0_address =
             Bytes::from_str("6B175474E89094C44Da98b954EedeAC495271d0F").expect("c0 address valid");
-        let exp_slots: HashMap<U256, U256> = vec![
-            (U256::from(0), U256::from(1)),
-            (U256::from(1), U256::from(5)),
-            (U256::from(2), U256::from(1)),
+        let exp_slots: HashMap<Bytes, Bytes> = vec![
+            (Bytes::from(0_u8).lpad(32, 0), Bytes::from(1_u8).lpad(32, 0)),
+            (Bytes::from(1_u8).lpad(32, 0), Bytes::from(5_u8).lpad(32, 0)),
+            (Bytes::from(2_u8).lpad(32, 0), Bytes::from(1_u8).lpad(32, 0)),
```
Comment (louise-poole) [inline tycho-storage/src/postgres/chain.rs:591]: I assume we'd need to do this combo very often (create left padded bytes). Maybe we could combine them into a single function? Maybe a new macro for a function like `into_left_padded_bytes` and we could do:
`0_u8.into_left_padded_bytes()`

Comment (louise-poole) [reply tycho-storage/src/postgres/chain.rs:591]: Hmmmm... going through the rest of this PR I see this isn't actually used anywhere else (I had expected it to be used often). We can ignore this comment above then 👍

Comment (zizou0x) [reply tycho-storage/src/postgres/chain.rs:591]: Actually it's used a lot, but I introduced a function especially for this case directly on the `Bytes` struct

**Pull request merged**